File: code/dobot_quad_sdk-main/high_level/cpp/e7_line_walk.cpp
// Example 7: Walk in a straight line or at an angle.
// Usage: ./e7_line_walk [server_address] [direction] [distance]
//   direction: 0=front (default), 1=back, 2=left, 3=right
//   distance:  meters (default 3.0)
//
// Simplified API:
//   client.line_walk(0, 3.0);        // front 3m
//   client.walk_forward(3.0);        // same thing
//   client.move_left(2.0);           // strafe left 2m
//   client.rotate_walk(-45, 2.0);    // walk at -45° (left turn)
//   client.rotate_walk(90, 1.0);     // walk at 90° (right turn)

#include "robot_client.h"

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    robot::enable_safety_ready(client);
    int dir = (argc > 2) ? std::stoi(argv[2]) : 0;
    float dist = (argc > 3) ? std::stof(argv[3]) : 3.0f;

    const char* names[] = {"front", "back", "left", "right"};
    std::cout << "Line Walk: " << names[dir] << " " << dist << "m" << std::endl;

    bool ok = client.line_walk(dir, dist);

    // Demonstrate rotate_walk with signed angles
    if (ok) {
        std::cout << "\nRotate Walk Demo:" << std::endl;
        std::cout << "  -45° (left turn): ";
        client.rotate_walk(-45, 1.0, 80, true);

        std::cout << "  90° (right turn): ";
        client.rotate_walk(90, 1.0, 80, true);
    }

    return ok ? 0 : 1;
}

File: code/dobot_quad_sdk-main/high_level/python/examples/e7_line_walk.py
#!/usr/bin/env python3
"""Example 7: Walk in a straight line.
Usage: python e7_line_walk.py [server_address] [direction] [distance]
  direction: 0=front (default), 1=back, 2=left, 3=right
  distance:  meters (default 3.0)
"""
import sys
from dobot_quad import RobotClient

DIRECTION_NAMES = ["front", "back", "left", "right"]


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    robot.enable_safety_ready()
    direction = int(sys.argv[2]) if len(sys.argv) > 2 else 0
    distance = float(sys.argv[3]) if len(sys.argv) > 3 else 1.0

    print(f"Line Walk: {DIRECTION_NAMES[direction]} {distance}m")
    # res = robot.line_walk(direction, distance)
    # rotate_walk convention: left is negative angle, right is positive angle.
    res = robot.rotate_walk(angle=-45, distance=1.0, speed_ratio=10)
    return 0 if res and res.success else 1


if __name__ == "__main__":
    sys.exit(main())
